### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by mrcuong84**

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,12 +1 @@
...
-      <div class="col-sm-3"><img src="https://media.istockphoto.com/vectors/bookstore-logo-template-design-logo-open-book-vector-id1088265598" alt="Img" class="avatar"></div>
-      <div class="col-sm-9">
-        <div class="row"><b class="custom-highlight-name"><%= user.name %></b></div>
-		    <div class="row"><b class="custom-highlight-name"><%= user.address %></b></div>
-		    <div class="row"><b class="custom-highlight-name"><%= user.phone %></b></div>
-      </div>
-    </div>
-  </div>
-<% end %>
+
```
Comment (dinhvareal) [inline app/views/users/index.html.erb]: Delete file

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,16 +1,33 @@
...
-      <h2>
-      	<%= @user.address %>
-      </h2>
-      <h3>
-      	<%= @user.phone %>
-      </h3>
-    </section>
-  </aside>
+  <div class="col-sm-1"></div>
+  <% if @user %>
```
Comment (dinhvareal) [inline app/views/users/show.html.erb]: Don't need because checking user available is existing in user_controller

File: config/locales/en.yml [hunk 3]
```diff
@@ -55,6 +55,10 @@ en:
               too_short: "password is too short"
             address:
               blank: "address is blank"
+            password_confirmation:
+              confirmation: "Wrong passwordaa cornfirmation"
```
Comment (dinhvareal) [inline config/locales/en.yml]: Are you okey? Hello. It should be "Wrong password confirmation"

File: config/locales/en.yml [hunk 4]
```diff
@@ -68,10 +72,12 @@ en:
       sign_in_now: "Sign In "
     create:
       success_login: "Success Logged In"
-
+      wrong_password: "Wrong password"
   users:
     show:
       user_not_exist: "User is not existed"
+      cap_nhat: "Update Infor"
```
Comment (dinhvareal) [inline config/locales/en.yml]: cap_nhat? Translate it to English

File: config/locales/en.yml [hunk 5]
```diff
@@ -82,7 +88,6 @@ en:
       create_my_account: "Create my account"
       input_guide_phone: "Format 09xx......"
       input_guide_password: "atlease 6 characters"
-
   shared:
     error_messages:
       errors_notification: "Form dang ky co %{errors_count} loi"
```
Comment (dinhvareal) [inline config/locales/en.yml]: Don't use VNese in en.yml

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+RSpec.describe User, type: :model do
+  context 'validations' do
+    it { should validate_presence_of(:name) }
+    it { should validate_presence_of(:address) }
+    it { should validate_presence_of(:password) }
+    it { should validate_presence_of(:phone) }
```
Comment (dinhvareal) [inline spec/models/user_spec.rb:7]: has_many book

File: test/factories/users.rb [hunk 7]
```diff
@@ -0,0 +1 @@
+
```
Comment (dinhvareal) [inline test/factories/users.rb]: Is it blank?

File: app/views/users/index.html.erb (hunk 1 above)
Comment (mrcuong84) [reply app/views/users/index.html.erb]: deleted

File: app/views/users/show.html.erb (hunk 2 above)
Comment (mrcuong84) [reply app/views/users/show.html.erb]: aready fixed

File: config/locales/en.yml (hunk 3 above)
Comment (mrcuong84) [reply config/locales/en.yml]: already fixed

File: config/locales/en.yml (hunk 4 above)
Comment (mrcuong84) [reply config/locales/en.yml]: fixed

File: config/locales/en.yml (hunk 5 above)
Comment (mrcuong84) [reply config/locales/en.yml]: fixed

File: spec/models/user_spec.rb (hunk 6 above)
Comment (mrcuong84) [reply spec/models/user_spec.rb:7]: fix when assign the book to user, later task

File: test/factories/users.rb (hunk 7 above)
Comment (mrcuong84) [reply test/factories/users.rb]: already removed

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -1,14 +1,19 @@
...
   def show
-    @user = User.find_by(id: params[:id])
-    return if @user.present?
+    if current_user
+      @user = User.find_by(id: params[:id])
+      return if @user.present?
 
-    flash[:danger] = t('.user_not_exist')
-    redirect_to new_user_path
+      @user = User.find_by(id: current_user.id)
```
Comment (hieunguyentrung) [inline app/controllers/users_controller.rb]: Current user can't see other users profile.

File: app/views/users/show.html.erb [hunk 9]
```diff
@@ -1,16 +1,34 @@
...
+    <div class="row">
+      <%= @user.address %>
+    </div>
+    <div class="row">
+      <%= @user.phone %>
+    </div>
+    <div class="row">
+      <p><%= t('.join') %> </p> <%= @user.created_at %>
+    </div>
+    <% if current_user.id == params[:id].to_i %>
```
Comment (hieunguyentrung) [inline app/views/users/show.html.erb]: Remove.

**Review by hieunguyentrung — CHANGES_REQUESTED** [review]

File: app/controllers/users_controller.rb (hunk 8 above)
Comment (mrcuong84) [reply app/controllers/users_controller.rb]: already fix

File: app/views/users/show.html.erb (hunk 9 above)
Comment (mrcuong84) [reply app/views/users/show.html.erb]: already fix

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -1,14 +1,11 @@
 class UsersController < ApplicationController
   def index
-    @user = User.all
+    redirect_to root_path
```
Comment (hieunguyentrung) [inline app/controllers/users_controller.rb]: Remove this `index` method. We don't need that.

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -1,14 +1,11 @@
...
   end
 
   def show
-    @user = User.find_by(id: params[:id])
+    @user = User.find_by(id: current_user.id)
```
Comment (hieunguyentrung) [inline app/controllers/users_controller.rb]: You already have `current_user`, why did you need to find it again ?

**Review by hieunguyentrung — CHANGES_REQUESTED** [review]

**Comment by hieunguyentrung** [comment]
Why do we need this file ?
`app/uploaders/logo.jpeg`

File: app/views/sessions/new.html.erb [hunk 12]
```diff
@@ -1,24 +1,24 @@
 <div class="row">
-  <h1><%=t('.login')%></h1>
-  <div class="col-md-6 col-md-offset-3">
+  <div class="col-md-4 col-md-offset-3">
+    <h1><%=t('.login')%></h1>
```
Comment (hieunguyentrung) [inline app/views/sessions/new.html.erb]: Need space after `=` and before `%`

File: app/views/sessions/new.html.erb [hunk 13]
```diff
@@ -1,24 +1,24 @@
...
       <%= f.submit t('.sign_in_now'), class: "btn btn-primary" %>
     </div>
-
     <% end %>
-
-    <%=t('.new_user')%> <%= link_to t('.sign_up_now'), new_user_path %>
+      <div class="row">
+        <div class="col-sm-1"></div>
+        <div class="col-sm-11">
+          <%=t('.new_user')%> &nbsp; <span class="underline"><%= link_to t('.sign_up_now'), new_user_path %></span>
```
Comment (hieunguyentrung) [inline app/views/sessions/new.html.erb]: Need space after `=` and before `%`

File: app/views/users/show.html.erb [hunk 14]
```diff
@@ -1,16 +1,31 @@
...
+    </div>
+      <div class="row">
+        <a href="#" class="btn btn-primary"><%= t('.update') %></a>
+      </div>
+  </div>
 </div>
+
+
+
+
```
Comment (hieunguyentrung) [inline app/views/users/show.html.erb]: A lot of blank lines.

**Review by hieunguyentrung — CHANGES_REQUESTED** [review]

**Review by hieunguyentrung — DISMISSED** [review]

**Review by hieunguyentrung — APPROVED** [review]
